### PR TITLE
Mock bee implementation

### PR DESCRIPTION
**Pull request by dev-prabhat**
Please check these files -
- Product-Context.js
- useAxios.js
- product.js in the backend
- category.js in the backend
-Filter-Context.js

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { v4 as uuid } from "uuid";
+
+/**
+ * Product Database can be added here.
+ * You can add products of your wish with different attributes
+ * */
+
+
+const getImg = (imageNumber) => process.env.PUBLIC_URL + `/images/img${imageNumber}.jpg`
```
Comment (sudipkundu999) [inline src/backend/db/products.js:9]: Appreciation: Good way of getting images from the public folder using a custom function

File: src/Context/Product-Context.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import React, { createContext, useState, useEffect, useContext } from "react"
+import { useAxios } from "../customHooks/useAxios"
+
+const ProductContext = createContext()
+
+const ProductProvider = ({ children }) => {
+    const [productsFromDB, setProductsFromDB] = useState([])
+    const [categoriesFromDB, setCategoriesFromDB] = useState([])
+    const { response, loading, operation: fetchProducts } = useAxios()
+    const { response: responseFromCategories, operation: fetchCategories } = useAxios()
```
Comment (sudipkundu999) [inline src/Context/Product-Context.js]: Try to follow a pattern for consistency and easy debugging.
You can also create an alias for `response` as `responseProducts` and `loading` as `responseLoading`

File: src/Context/Filter-Context.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import React, { createContext, useContext, useReducer } from "react"
-import ProductData from "../Data/ProductData"
+import { useProduct } from "./Product-Context"
 
 const FilterContext = createContext()
 
 const FilterProvider = ({ children }) => {
+    const { productsFromDB } = useProduct()
+    // console.log(productsFromDB)
```
Comment (sudipkundu999) [inline src/Context/Filter-Context.js]: remove console logs

**Review by sudipkundu999 — APPROVED** [review]
Good to merge with changes.

File: src/index.js [hunk 4]
```diff
@@ -5,18 +5,24 @@ import App from './App';
 import { CartProvider } from './Context/Cart-Context';
 import { FilterProvider } from './Context/Filter-Context';
 import { WishlistProvider } from './Context/Wishlist-Context';
+import { ProductProvider } from "./Context/Product-Context"
 import reportWebVitals from './reportWebVitals';
+import { makeServer } from "./server"
```
Comment (bhushannaikaroskar) [inline src/index.js:10]: Hey prabhat maybe you can use one single file index.js to import all your context at once.

**Review by bhushannaikaroskar — APPROVED** [review]
Hey prabhat nice implementation. I would suggest you to do the required changes.

**Review by sach10-create — COMMENTED** [review]
Good to merge

**Pull request merged**